### ISSUE TITLE
Update Cupertino widgets library

### ISSUE THREAD

**Issue by Renesanse** [flutter/flutter#102027]
Cupertino lib is defently out of date.

**Comment by Shawn-sudo** [comment]
I also agree. Besides the appearance, there are also some features like "moving finger around buttons" that is supported on iOS, but couldn't find the feature on Flutter:

https://user-images.githubusercontent.com/60294318/163690534-d23f5ebe-df23-4533-9958-83d6e9ef5d7b.mov

**Comment by Shawn-sudo** [comment]
Oh how about just using the native iOS api to update some of the cupertino widgets (such as CupertinoAlertDialog, CupertionActionSheet)?

Pros
 - Easy to implement (at least it seems like)
   - Apple seems to change their design a lot, and it's hard to keep catching those up in a Flutter-way, which is basically "emulating" or "reverse-engineering" those. Just using native api can be an option.
   - Apple is such a pervert
     - There is always some "hidden feature" and I think quite many users (including me!) enjoy those. It would be easier to implement those hidden feature using the native api.
 - Performance
    - I'm thinking of the Cupertino widgets not to deal with drawing widgets, and only to deal with functional things.
      - What I'm saying is like this: On input, Cupertino widgets would call `open contextMenu here` or `show IOS native dialog here and call this func when some menu is selected` rather than calling `change positionX to 1.2, positionY to 3.4` `On the next frame, change positionX to 2.4 and positionY to 5.6`

Cons of using native API
 - Fragmentation (possibly, not necessarily)
 - It wouldn't be able to implement every Cupertino widgets using Native API: eg. CupertinoPageRoute highly depends on Flutter's Navigation management thingy, so I think it would be hard to use the native one I think.


I do have some experience developing with pure Flutter, but I have zero experience connecting Flutter and the native OS, so if my idea doesn't make any sense, well that'll be so sad.

**Comment by darshankawar** [comment]
@Renesanse 
Check [this](https://github.com/flutter/flutter/issues/72926) similar issue and see if tracking in it would be helpful and answer your questions.

**Comment by notandyvee** [comment]
>  Cupertino routes are laggy

I was experiencing janky routes as well. Had to hack around it. Namely the swipe right to go back. This seems to be a core problem, and not a widget problem. Anyone have an explanation as to the jank on iOS navigation animations?

**Comment by Renesanse** [comment]
> > Cupertino routes are laggy
> 
> I was experiencing janky routes as well. Had to hack around it. Namely the swipe right to go back. This seems to be a core problem, and not a widget problem. Anyone have an explanation as to the jank on iOS navigation animations?

try wrap in repaintBoundary child inside route(or builder) it is fix the problem. But I sure it should be by default.

**Comment by darshankawar** [comment]
> @Renesanse Check [this](https://github.com/flutter/flutter/issues/72926) similar issue and see if tracking in it would be helpful and answer your questions.

@Renesanse Did you get a chance to look at linked issue and would be ok to track in it for your original request ?
